sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1086 @@
+     * the frame.
+     */
+    IEEE802154_RADIO_INDICATION_RX_DONE,
+
+    /**
+     * @brief the transceiver finished sending a frame or the
+     *        retransmission procedure
+     *
+     * If the radio supports frame retransmissions the 
+     * @ref ieee802154_radio_get_tx_status MAY be called to retrieve useful
+     * information (number of retries, frame pending bit, etc). The
+     * transceiver is in @ref IEEE802154_TRX_STATE_TX_ON state when this function
+     * is called.
+     */
+    IEEE802154_RADIO_CONFIRM_TX_DONE,
+    /**
+     * @brief the transceiver reports that the ACK timeout expired
+     *
+     * This event is present only if the radio support ACK timeout.
+     */
+    IEEE802154_RADIO_CONFIRM_ACK_TIMEOUT,
+    /**
+     * @brief the CCA procedure finished
+     */
+    IEEE802154_RADIO_CONFIRM_CCA,
+} ieee802154_trx_ev_t;
+
+/**
+ * @brief CSMA-CA exponential backoff parameters.
+ */
+typedef struct {
+    uint8_t min; /**< minimum value of the exponential backoff */
+    uint8_t max; /**< maximum value of the exponential backoff */
+} ieee802154_csma_be_t;
+
+/**
+ * @brief RX information associated to a frame
+ */
+typedef struct {
+    int16_t rssi; /**< RSSI of the received frame */
+    uint8_t lqi;  /**< LQI of the received frame */
+} ieee802154_rx_info_t;
+
+/**
+ * @brief Transmission flags.
+ */
+typedef enum {
+    /**
+     * @brief transmission uses Auto CCA or frame retransmissions with CSMA-CA
+     */
+    TX_MODE_AUTO=(1<<0),
+    /**
+     * @brief the transceiver goes immediately to RX state after TX done.
+     */
+    TX_MODE_TXRX=(1<<1),
+} ieee802154_tx_flags_t;
+
+/**
+ * @brief TX information of the last transmitted packet.
+ */
+typedef struct {
+    ieee802154_tx_flags_t tx_flags; /**< flags of the last transmission */
+    int8_t retries;                 /**< number of frame retransmissions of the last TX */
+} ieee802154_tx_info_t;
+
+/**
+ * @brief Forwared declaration of the IEEE802.15.4 device descriptor
+ */
+typedef struct ieee802154_dev ieee802154_dev_t;
+
+/**
+ * @brief Prototype of the IEEE802.15.4 device event callback
+ *
+ * @param[in] dev IEEE802.15.4 device descriptor
+ * @param[in] status the status 
+ */
+typedef void (*ieee802154_cb_t)(ieee802154_dev_t *dev,
+                                ieee802154_trx_ev_t status);
+
+/**
+ * @brief the IEEE802.15.4 device descriptor
+ */
+struct ieee802154_dev {
+    /**
+     * @brief pointer to the operations of the device
+     */
+    const ieee802154_radio_ops_t *driver;
+    /**
+     * @brief the event callback of the device
+     */
+    ieee802154_cb_t cb;
+};
+
+/**
+ * @brief IEEE802.15.4 CCA modes
+ */
+typedef enum {
+    IEEE802154_CCA_MODE_ED_THRESHOLD,     /**< CCA mode 1 */
+    IEEE802154_CCA_MODE_CARRER_SENSING,   /**< CCA mode 2 */
+    IEEE802154_CCA_MODE_ED_THRESH_AND_CS, /**< CCA mode 3 (AND) */
+    IEEE802154_CCA_MODE_ED_THRESH_OR_CS,  /**< CCA mode 3 (OR) */
+} ieee802154_cca_mode_t;
```

Comment:
The description of modes can be precised. Or is "CCA mode x" a common term?

Reply:
it's a common term defined by the standard.

Reply:
Would be nice to have a short description of what that implies - not everyone always has the standard handy :wink:

Reply:
Then it should be fine I guess...

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/30d9f00c7a3b6dbaaf7aa15a8183ac34ad768060